ci/mkpipeline.py

```diff
@@ -289,7 +291,16 @@ def prioritize_pipeline(pipeline: Any, priority: int) -> None:
         priority -= 10
 
     def visit(config: Any) -> None:
-        config["priority"] = config.get("priority", 0) + priority
+        # Increase priority for larger Hetzner-based tests so that they get
+        # preferential treatment on the agents which also accept smaller jobs.
+        agent_priority = 0
+        if "agents" in config:
+            agent = config["agents"].get("queue", None)
+            if agent == "hetzner-aarch64-8cpu-16gb":
+                agent_priority = 1
+            if agent == "hetzner-aarch64-16cpu-32gb":
+                agent_priority = 2
+        config["priority"] = config.get("priority", 0) + priority + agent_priority
```

Comment:
Where is the size of tests ("for larger Hetzner-based tests") in this computation?

Reply:
Line 299-302. This is basically about the 4cpu (prio 0)/8cpu (prio 1)/16cpu (prio 2) agents, and is kind of a micro-optimization since we can't provision enough servers at the moment. Without this change sometimes a 4cpu job would grab a 16cpu agent, making it unavailable for the 16cpu job that was waiting at the same time. (I was really hoping to not require all this complexity previously.)